### ISSUE TITLE
Notification once barcode is scanned.

### ISSUE THREAD

**Issue by irperera** [AmolGangadhare/flutter_barcode_scanner#32]
**Is your feature request related to a problem? Please describe.**
Once Streaming using the getBarcodeStreamReceiver, there should be notification bar or snackbar that popup and let u know the barcode was scan and press ok to continue to scan.

**Describe the solution you'd like**
have a snackbar popup

**Describe alternatives you've considered**

**Additional context**

**Comment by lookfirst** [comment]
I can't imagine why anyone would want to tap an OK button after scanning something. Imagine scanning 1000 things... that is 1000 OK taps. Even still, you could add this yourself to your own project, you don't need to add it to this project.

**Comment by irperera** [comment]
so how will you know u actually scan the item? There needs to be a way to when it scan a card can some up for a bit and go back done.

**Comment by lookfirst** [comment]
It depends on your UX. 

In my app, I have a set number of Things that need to be scanned. So, I have a screen with one button for each Thing. User clicks the button, scans and when done is shown the screen again. The button goes from yellow to green to show that the item has been scanned and displays the scanned value of the Thing just below the button (it is a serial number). I also have haptic feedback to let the person know the scan is completed without even looking at the screen. This UX will be used to scan about 300k+ QR codes, so I can't imagine displaying a modal they have to click OK on as that is a lot of clicks.

I'm sorry that you disagree above with my advice, but why don't you ask a few other people or do some usability studies? I'm sure they will tell you the same thing when they are scanning a whole lot of items.

**Comment by lookfirst** [comment]
To do it on your own, simply display a modal once you get a result back from the scan. That should be pretty easy and does not require any changes to this project.

**Comment by irperera** [comment]
I'm having issues with the model keeping on top of the camera screen. if you can, would you be able to show me the process on your code?

**Comment by lookfirst** [comment]
Ah, I missed the part where you are streaming. I'd say that the model isn't staying 'on top' because that is a bad idea to do that.

The streaming use case requires another screen. It would be that you have the device always scanning and the results are showing up on another screen. Just like a hand scanner.

Which I don't understand why anyone would buy an expensive mobile phone and leave it in streaming mode, when a less expensive (and more durable/reliable) hand scanner is the right solution.

**Comment by irperera** [comment]
what is your recommendation?

how can I call the function over?

**Comment by lookfirst** [comment]
I gave you my recommendation. If you want to do streaming, buy a hand scanner.

**Comment by dashcg** [comment]
I was going to use your plugin with continuous scan to create an event ticket scanning app. The ability to continually scan and see the result of the ticket scan being a valid ticket would be helpful. Save time for our volunteers at the event not having to press a button, simply pass the ticket barcode under the camera and see a result of the scan after passing it to the server for validation.

**Comment by irperera** [comment]
I completely agree with you. Its the same thing I'm looking for, I had to use a hack for it to work with a timer.

**Comment by dashcg** [comment]
I'm using this hack to fake a continuous scan. Probably similar to what you're doing as well. [https://github.com/AmolGangadhare/flutter_barcode_scanner/issues/18#issuecomment-528213059](url)

**Comment by lookfirst** [comment]
@dashcg 

1. Scan
1. Show 'green' confirmation screen for X seconds.
1. Step 1.

You don't need a continuous scan. It is just a loop.

**Comment by git-xbits** [comment]
I really need this soluction too. Unfortunatelly I work with tickets check, so I need to get a confirmation and check next very quickly.

**Comment by git-xbits** [comment]
> @dashcg
> 
>     1. Scan
> 
>     2. Show 'green' confirmation screen for X seconds.
> 
>     3. Step 1.
> 
> 
> You don't need a continuous scan. It is just a loop that you can break by hitting cancel in the scan screen.

How do you make this ? Unfortunatelly I can't .

**Comment by lookfirst** [comment]
@git-xbits Create an example project and let's see what you have. It really shouldn't be hard to start a scan and then get a value and repeat the scan. As soon as you don't get a value (due to hitting cancel), you stop.

But really... for something like checking a huge number of tickets, it is FAR better to use a dedicated scanning device over a mobile phone. You're asking for trouble doing what you're doing.

**Comment by irperera** [comment]
@lookfirst, if you look at other apps, example target, they have the barcode scanning where by you scan it pops up with a card that said the item was found, which goes away and u can continue scanning again without it calling a new instance of the camera.

**Comment by irperera** [comment]
@git-xbits and @dashcg 

`Future<void> pressScan() async {
  var barcode = await _getBarcode();
  if (barcode.isNotEmpty) {

    setState(() {
      result = barcode;
    });

    var list = //Your Data CodeHere

    var foundItem =
    list.where((product) => product.productBarcode == result).toList();

    if (foundProduct.length == 0) {
      Flushbar(
        title: "Not Found",
        message: "Scanned Product Not Found",
        duration: Duration(seconds: 2),
        backgroundColor: Colors.red,
      ).show(context);
    } else {
        Flushbar(
          title: "Added to Cart",
          message: "#${item.id} Has Been Added To Cart",
          duration: (Duration(seconds: 2)),
          backgroundColor: Colors.green,
        ).show(context);
      }
    }
  }
  //I have a Toggle in my settings screen to turn on or off continuous Scanning.
  if (Utils.prefs.getBool("BarcodeGo") == true) {
    //I user a timer, which allows the Popup to Show before starting the loop.
    Timer(Duration(seconds: 3), () {
      if (barcode.isNotEmpty) pressScan();
    });
  }
}`

`Future<String> _getBarcode() async {
  String barcodeScanResult;
  try {
    barcodeScanResult = await FlutterBarcodeScanner.scanBarcode(COLOR_CODE, CANCEL_BUTTON_TEXT, isShowFlashIcon, scanMode);
  } on PlatformException catch (e) {
    if (e.code == BarcodeScanner.CameraAccessDenied) {
      setState(() {
        result = "Camera permission was denied";
      });
    } else {
      setState(() {
        result = "Unknow Error $e";
      });
    }
  } on FormatException {
    setState(() {
      result = "You pressed the back button before scanning.";
    });
  } catch (e) {
    setState(() {
      result = "Unknow Error $e";
    });
  }
  return barcodeScanResult ?? "";
}
`

**Comment by AmolGangadhare** [comment]
Most of the users are seems to need this thing.

I'm planning to add some solution to this.

Have a boolean flag e.g. `scanConfirmationRequired` based on which the alert will be shown.

`true` = show alert confirmation
`false` = do not show alert confirmation.

Let's have some feedback on this.

**Comment by thomahart** [comment]
How about a beep as an option after which the barcode was successfully detected ?

**Comment by limenote135** [comment]
In my case, I don't want to show confirmation popup but notify read result by toast or snack bar.
Currently, I think users cannot know the barcode was read or not yet, right?

**Comment by Gaffen** [comment]
Yeah, similarly for me it would be nice to inform the user when an item has successfully scanned in continuous mode.

**Comment by ScorprocS** [comment]
It would be greate if we could use the FlutterBarcodeScanner.getBarcodeStreamReceiver context to show whatever we want on success or on failure reading barcode and search for the code on database.
Dispite that I really thing this is an awsome library.

**Comment by ClementVincent** [comment]
I'm also looking for this feature. There is currently no feedback on the stream scan so the user cannot know if the barcode has been properly scanned. 
But in my opinion, adding a popup will go against the general purpose of the streaming scan which is to scan multiple barcode in a short time (without adding extra interactions).
Although @ScorprocS solution would be great, maybe just a check icon appearing on top of the scanner would suffice. Like with a duration parameter to avoid the scanner to scan the barcode while the icon is displayed.

Thank you anyway for you work :)

**Comment by legeion** [comment]
> Que diriez-vous d'un bip en option après lequel le code-barres a été détecté avec succès?



> How about a beep as an option after which the barcode was successfully detected ?

Good morning all.
I want to thank the designer of the package; it is a very good job which helped me a lot in my personal project.

Happy New Year to everyone.

I bring this proposal back to see if anyone could have done it.
since when you do continuous scans, you don't know if it's taken into account or not?
suddenly we have trouble using it
